SmartCart.DataProvider/Repositories/CartRepository/CartRepository.cs

```diff
@@ -0,0 +1,119 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using SmartCart.DataProvider.DatabaseContext;
+using SmartCart.DataProvider.Models;
+
+namespace SmartCart.DataProvider.Repositories
+{
+    public class CartRepository : ICartRepository
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CartRepository(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<bool> AddAsync(CartDto cart)
+        {
+            if(cart == null)
+            {
+                return false;
```

Comment:
Raise an exception, as program now in suspicious and unpredictable state. Bad practice.